public/js/voltaLogin.js

```diff
@@ -0,0 +1,59 @@
+// window.onbeforeunload = function (event) {
+//     window.location.replace(document.referrer);
+// };
+
+(function (window) {
+    'use strict';
+
+    var noback = {
+
+        //globals 
+        version: '0.0.1',
+        history_api: typeof history.pushState !== 'undefined',
+
+        init: function () {
+            window.location.hash = '#no-back';
+            noback.configure();
+        },
+
+        hasChanged: function () {
+            if (window.location.hash == '#no-back') {
+                window.location.hash = '';
+                //mostra mensagem que não pode usar o btn volta do browser
+            }
+        },
+
+        checkCompat: function () {
+            if (window.addEventListener) {
+                window.addEventListener("hashchange", noback.hasChanged, false);
+            } else if (window.attachEvent) {
+                window.attachEvent("onhashchange", noback.hasChanged);
+            } else {
+                window.onhashchange = noback.hasChanged;
+            }
+        },
+
+        configure: function () {
+            if (window.location.hash == '#no-back') {
+                if (this.history_api) {
+                    history.pushState(null, '', '');
+                } else {
+                    window.location.hash = '';
+                }
+            }
+            noback.checkCompat();
+            noback.hasChanged();
+        }
+
+    };
+    if (typeof define === 'function' && define.amd) {
```

Comment:
Toda essa estrutura de decisão é desnecessária, não usamos nem `amd` nem `module.exports`. Podemos apagar  of `if` e o `else if`, podemos deixar somente a linha de dentro do `else`.